lib/rubocop/cop/style/hash_syntax.rb

```diff
@@ -96,7 +96,13 @@ def word_symbol_pair?(pair)
           return false unless key.sym_type?
 
           sym_name = key.loc.expression.source
-          sym_name !~ /\A:["']|=\z/
+          valid_19_syntax_symbol?(sym_name)
+        end
+
+        def valid_19_syntax_symbol?(sym_name)
+          return false if sym_name =~ /\A:["']/
```

Comment:
I think you’ll have to ask @bbatsov this. Note that [`ProcessedSource` uses `Parser::CurrentRuby`](https://github.com/bbatsov/rubocop/blob/v0.30.0/lib/rubocop/processed_source.rb#L95), so in theory this should return the valid syntax for the given version of Ruby.